NMS-14350: Add terms to glossary.adoc (#4893)
OpenNMS/opennms: docs/modules/reference/pages/glossary.adoc

> (See xref:operation:provisioning/auto-discovery.adoc[Auto-Discovery].)

Reviewer: Given earlier discussion, we could remove this link and make the heading the link.

Availability:: The percentage of time a node has been detected as being accessible.
{page-component-title} typically shows availability statistics for the past 24 hours.
This is calculated by subtracting the amount of outage time for a node, interface, or service during the time period.

> Azure:: Microsoft Azure is a cloud hosting platform that allows users to host a variety of different services.

Reviewer: ```suggestion
Azure:: Microsoft Azure is a cloud hosting platform that allows users to host different services.
```
"Variety" and "different" are redundant.

Bean Scripting Framework (BSF):: A framework that allows scripting to be used in Java code via a set of Java classes.
The {page-component-title} xref:reference:provisioning/detectors/BsfDetector.adoc[BSF Detector] runs arbitrary BSF scripts to determine the existence of a service, and the xref:reference:service-assurance/monitors/BSFMonitor.adoc[BSF Monitor] uses scripts to determine the status of a service.

Cassandra:: An open-source distributed/scale-out NoSQL database designed to hold large amounts of structured or unstructured data across many servers.

> The optional Newts storage strategy requires a Cassandra instance as back-end storage.

Reviewer: Make "Newts" a link and remove the link below.

Author: Done in VSC and will be committed with those changes.

(See xref:operation:provisioning/auto-discovery.adoc[Auto-Discovery].)

Availability:: The percentage of time a node has been detected as being accessible.

> {page-component-title} typically shows availability statistics for the past 24 hours; this is calculated by subtracting the amount of outage time for a node, interface, or service.

Author: I don't think we need to say "for the given time period" since we're talking about a period of 24 hours being shown. If we do want to break this into two sentences, though, maybe the below?

```suggestion
{page-component-title} typically shows availability statistics for the past 24 hours.
This is calculated by subtracting the amount of outage time for a node, interface, or service during the time period.
```

> {page-component-title} typically shows availability statistics for the past 24 hours; this is calculated by subtracting the amount of outage time for a node, interface, or service.

Reviewer 2: I like it.

Direct-discovered nodes:: Nodes that are added to {page-component-title} manually through a requisition.
(See xref:operation:provisioning/directed-discovery.adoc[Directed Discovery].)

Discourse:: An open-source forum platform that hosts https://opennms.discourse.group/[OpenNMS's community forums].
There, users can find knowledge base articles, support, announcements, and general discussion about all things OpenNMS.

Discovery:: A process that uses ICMP to find network devices and services in your network and then automatically adds them to monitoring.

> (See xref:operation:provisioning/introduction.adoc#discovery-auto[Provisioning].)

Reviewer: > Thoughts on this format of making the term a link instead of having the `(See ...)` at the end?

Hmm. That's a possibility. Not all the terms in the glossary will have links, so it might look odd. Also, there are some `(See ...)` links that aren't to the actual glossary term, but something related to it. 

The purpose of the link is to provide additional information beyond the defined term. Should we replace the `(See ..)` links that are for the same term with "Deep dive" or "More info" and a link, and keep the ones with related topics as `(See also ...)?

> (See xref:operation:provisioning/introduction.adoc#discovery-auto[Provisioning].)

Reviewer 2: I think the current :"See..." terminology for those items are fine.  Was just thinking that if the text of the reference matches the term or a phrase in the definition, it would be more concise to make the term/phrase a link. At least for xrefs to internal pages.  Anything that's a link outside the sites we control should stay as is.

> (See xref:operation:provisioning/introduction.adoc#discovery-auto[Provisioning].)

Reviewer: And making only some terms links seems to be acceptable within best practices for online glossaries.

(See xref:operation:provisioning/getting-started.adoc#foreign-source-definition[Foreign Source Definition].)

Geolocation:: The actual physical location of a node monitored by {page-component-title}.
Nodes can be manually assigned a location, or the location can be automatically determined using the xref:reference:provisioning/adapters/geoip.adoc[GeoIP Provisioning Adapter].

Grafana:: An open-source analysis and visualization web application.
It connects to data sources and generates dashboards with charts, graphs, and alerts.
(See https://grafana.com/[Grafana].)
OpenNMS HELM is a Grafana plugin that allows users to create customized dashboards with data from {page-component-title}.
(See https://docs.opennms.com/helm/latest[HELM Documentation].)

HELM:: An OpenNMS application for creating flexible dashboards to display and customize fault, flow, and performance data from {page-component-title}.
HELM works with the Grafana analytics program to customize the data you can view.
Not to be confused with the https://helm.sh/[Kubernetes Helm] package manager.
(See https://docs.opennms.com/helm/latest[HELM Documentation].)

> Horizon:: An open-source solution from OpenNMS that helps users visualize and monitor everything on their local and remote networks.

Reviewer 2: Could potentially link Horizon/Meridian to the product pages on the web site.

> Horizon:: An open-source solution from OpenNMS that helps users visualize and monitor everything on their local and remote networks.

Author: ```suggestion
Horizon:: An open-source solution from The OpenNMS Group that helps users visualize and monitor everything on their local and remote networks.
(See https://www.opennms.com/horizon/[OpenNMS Horizon].)
```

Not to be confused with the https://helm.sh/[Kubernetes Helm] package manager.
(See https://docs.opennms.com/helm/latest[HELM Documentation].)

Horizon:: An open-source solution from The OpenNMS Group that helps users visualize and monitor everything on their local and remote networks.

> (See https://www.opennms.com/horizon/[OpenNMS Horizon].)

Reviewer: Let's remove the reference here and make the header the link. Could you add a similar link to the Meridian entry, but to https://www.opennms.com/meridian/ ?

(See https://issues.opennms.com[OpenNMS Jira].)

JRobin:: A clone of RRDTool (see below) written in Java.
Default time series database for new {page-component-title} instances.

Kafka:: Clusterable, open-source message broker designed for high-throughput/low-latency event streaming.
Originally developed at LinkedIn, it is now supported by the Apache community.
(See xref:deployment:core/setup-message-broker.adoc[Set Up Message Broker].)

Kibana:: A component of the Elastic Stack (comprised of Elasticsearch, Logstash, and Kibana).
It provides visualization capabilities for data indexed in an Elasticsearch cluster.

KSC Reports:: Key SNMP customized (KSC) reports provide a way to generate prefabricated graphical views of collected data.
They let you display data from different devices and sources (SNMP, ICMP, HTTP) on one page.
(See xref:operation:admin/webui/opsboard/dashlet/ksc.adoc#ksc[KSC Reports].)

Kubernetes:: An open-source container orchestration system for automating software deployments, scaling, and management.
Originally designed by Google, the Cloud Native Computing Foundation now maintains it.
(See https://kubernetes.io/[Kubernetes].)

> Lightweight Directory Access Protocol (LDAP):: An industry standard protocol used by {page-component-title} to access and maintain distributed directory information services on an IP network.

Author: @mmahacek do you think this term should be included or removed? It seems to me that OpenNMS is using LDAP pretty much as it's generally defined, based on the documentation I read through.

> Lightweight Directory Access Protocol (LDAP):: An industry standard protocol used by {page-component-title} to access and maintain distributed directory information services on an IP network.

Reviewer 2: I wouldn't complain about removal.

> Lightweight Directory Access Protocol (LDAP):: An industry standard protocol used by {page-component-title} to access and maintain distributed directory information services on an IP network.

Author: ```suggestion
```

Nephron:: OpenNMS component that enables horizontal scaling of flow processing.
Not required for flow processing, but can improve performance for deployments with a very high volume of flow documents.

Network Operating Center (NOC):: One or more locations where a network is monitored and controlled.
(See https://en.wikipedia.org/wiki/Network_operations_center[Network Operations Center].)

Newts:: Newts (New-Fangled Time Series Datastore) is an OpenNMS time series data store based on Apache Cassandra.
Use it as an alternative persistence strategy instead of JRobin or RRDtool.
(See xref:deployment:time-series-storage/newts/introduction.adoc#ga-opennms-operation-newts[Newts].)

> Northbounder:: An alarm generated by {page-component-title} that is sent to a higher-level component (service, resource, etc.).

Reviewer 2: I'm not an expert on northbounding, but my perception of it was sending an object from one Horizon/Meridian instance to another, external system, whether it's another H/M instance or something else.

Author: I'll ask for clarification on this one. :)

You can build requisitions iteratively and import them at a later date.
(See xref:operation:provisioning/getting-started.adoc#requisition-create[Create a Requisition].)

Resource:: Any hardware or software that {page-component-title} can access.

> Round Robin Database (RRD):: A special type of database designed for collection, visualization, and analysis of time-series data.

Reviewer: ```suggestion
Round Robin Database (RRD):: A database designed for collection, visualization, and analysis of time-series data.
```

(See xref:operation:provisioning/getting-started.adoc#provision-snmp-configuration[Configure SNMP for Provisioning].)

Snaps:: Self-contained software packages that run in a sandbox and have mediated access to host systems.
Our Appliance Service uses snaps to distribute operating system packages and Minion service updates.

Telemetry:: The process of automatically recording and transmitting data from an external source to a system in a different location for monitoring or analysis.
The xref:operation:telemetryd/introduction.adoc[Telemetry daemon] in {page-component-title} accepts data sent from nodes in one of the supported protocol formats.

Time series, time-series database (TSDB):: Time series is a sequence of data points that occur in successive order over a period of time.
A time series database (TSDB) is designed to store and serve time series data.
(See xref:deployment:time-series-storage/timeseries/time-series-storage.adoc[Time Series Storage] and https://en.wikipedia.org/wiki/Time_series_database[Time Series Database].)

> xref:operation:topology/introduction.adoc[Topology]:: A description of a network's elements (devices, services) and the connections between them.

Reviewer: ```suggestion
xref:operation:topology/introduction.adoc[Topology]:: A description of a network's elements (devices, services) and the connections among them.
```

You can build requisitions iteratively and import them at a later date.
(See xref:operation:provisioning/getting-started.adoc#requisition-create[Create a Requisition].)

Resource:: Any hardware or software that {page-component-title} can access.

Round Robin Database (RRD):: A special type of database designed for collection, visualization, and analysis of time-series data.
Data is maintained as a fixed-size circular-buffer that overwrites the oldest data with new data.

Sample:: A small set of data collected from a {page-component-title}-monitored source.

Sentinel:: Karaf container that provides scalability for data processing of flow data.
It also supports thresholding for streaming telemetry if you are using the xref:deployment:time-series-storage/newts/introduction.adoc[Newts] time-series strategy.
(See xref:deployment:sentinel/introduction.adoc[Sentinel].)

> Service:: Software that performs a specified set of tasks (automation, listening for data requests, etc.).
> 

Author: ```suggestion
```

Simple Mail Transfer Protocol (SMTP):: An internet standard protocol.
{page-component-title} monitors SMTP availability on network nodes to ensure that email messages can be sent and received.
(See xref:reference:service-assurance/monitors/MailTransportMonitor.adoc[MailTransportMonitor].])

> Simple Network Management Protocol (SNMP):: An internet standard protocol used by {page-component-title} to collect, organize, and modify information about managed devices on an IP network.

Reviewer 2: Don't need the middle line, it's implied by having the link to configure.

```suggestion
Simple Network Management Protocol (SNMP):: An internet standard protocol used to collect, organize, and modify information about managed devices on an IP network.
(See xref:operation:provisioning/getting-started.adoc#provision-snmp-configuration[Configure SNMP for Provisioning].)
```

Snaps:: Self-contained software packages that run in a sandbox and have mediated access to host systems.
Our Appliance Service uses snaps to distribute operating system packages and Minion service updates.

> Southbounder:: An alarm generated by {page-component-title} that is sent to a lower-level component (service, resource, etc.).

Reviewer 2: First time I've heard this term used. 😀

Reviewer 2: the existence of "northbound" implies the existence of "southbound"
https://github.com/OpenNMS/opennms/blob/7e9f5759badaf5636ae487a72b605a2a89e3e13e/opennms-alarms/api/src/main/java/org/opennms/netmgt/alarmd/api/support/AbstractNorthbounder.java
> AbstractNorthBounder
<p>The purpose of this class is manage the queue of alarms that need to be forward and receive queries to/from a Southbound Interface.</p>
<p>It passes Alarms on to the forwardAlarms method implemented by base classes in batches as they are added to the queue.  The forwardAlarms method does the actual work of sending them to the Southbound Interface.</p>
<p>Preserve, accept and discard are called to add the Alarms to the queue as appropriate.</p>

Author: Looks like we don't have anything called a Southbounder within the products, and it's not a term used often, so I think we can leave this one out for now. :)
